### PR TITLE
Update Microsoft.Extensions.Azure for Workload Identity Support in Storage Utilities

### PR DESCRIPTION
**Pull request by wsugarman**
`Microsoft.Extensions.Azure` version `1.7.0` includes [support for the "credential type" value `"workloadidentity"`](https://github.com/Azure/azure-sdk-for-net/blob/Microsoft.Extensions.Azure_1.7.0/sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md). Updating this package will allow other extensions, like Timer, to easily enable workload identity authentication

Related to https://github.com/Azure/azure-functions-host/issues/9266

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Host.Storage/WebJobs.Host.Storage.csproj [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
+
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
     <StyleCopTreatErrorsAsWarnings>false</StyleCopTreatErrorsAsWarnings>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <WarningsAsErrors />
-  </PropertyGroup>  
+  </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Azure.Storage.Blobs" Version="12.13.0" />
```
Comment (wsugarman) [inline src/Microsoft.Azure.WebJobs.Host.Storage/WebJobs.Host.Storage.csproj:28]: I can also update this package too to latest while I'm here

**Review by aishwaryabh — APPROVED** [review]
LGTM!

**Comment by brettsam** [comment]
@wsugarman -- just realized why the tests are failing. DevOps won't allow secret access for accounts that are not part of our team. Seeing if there's a good way to get these tests run... please hold.

**Comment by wsugarman** [comment]
Changes included in part of https://github.com/Azure/azure-webjobs-sdk/pull/3046

**Pull request closed without merging**